File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt = 0 , el;
        for( int i = 0 ; i < nums.size() ; i++ ){
            if(cnt == 0)
                el = nums[i];
            
            if(el == nums[i])
                cnt++;
            else
                cnt--;
        }
        cnt = 0;
        for(auto i : nums){
            if(i == el)
                cnt++;
        }

        if(cnt > nums.size()/2)
            return el;
        else
            return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int maxEl = nums[0] ;
        int cnt = 1 ;
        for(int i = 1 ; i < nums.length ; i++){
            if(maxEl == nums[i])
                cnt++ ;
            else
                cnt-- ;

            if(cnt == 0){
                maxEl = nums[i] ;
                cnt = 1;

            }
        }
        cnt = 0 ;
        for(int i = 0 ; i < nums.length ; i++){
            if(maxEl == nums[i])
                cnt++;
        }
        if(cnt >= nums.length/2)
            return maxEl ;
        return -1 ;
    }
}